Tribler/Core/Modules/restapi/my_channel_endpoint.py

```diff
@@ -88,64 +103,154 @@ def render_PUT(self, request):
+    def render_PUT(self, request):
+        """
+        Add a torrent file to your own channel.Returns error 500 if something is wrong with the torrent file
+        and DuplicateTorrentFileError if already added to your channel.
+
+        Example request:
+        {
+            "torrent": "base64 encoded string of torrent file contents",
+            "description" (optional): "A video of my cat" (default: empty)
+        }
+        """
+        my_channel_id = self.channel_db_handler.getMyChannelId()
+        if my_channel_id is None:
+            return MyChannelBaseEndpoint.return_404(request)
+
+        parameters = http.parse_qs(request.content.read(), 1)
+
+        if 'torrent' not in parameters or len(parameters['torrent']) == 0:
+            request.setResponseCode(http.BAD_REQUEST)
+            return json.dumps({"error": "torrent parameter missing"})
+
+        if 'description' not in parameters or len(parameters['description']) == 0:
+            extra_info = {}
+        else:
+            extra_info = {'description': parameters['description'][0]}
+
+        try:
+            torrent = base64.b64decode(parameters['torrent'][0])
+            torrent_def = TorrentDef.load_from_memory(torrent)
+            self.session.add_torrent_def_to_channel(my_channel_id, torrent_def, extra_info, forward=True)
+
+        except (DuplicateTorrentFileError, ValueError) as ex:
+            return MyChannelBaseEndpoint.return_500(self, request, ex)
+
+        return json.dumps({"added": True})
+
+
+class MyChannelModifyTorrentsEndpoint(MyChannelBaseEndpoint):
+    """
+    This class is responsible for methods that modify the list of torrents (adding/removing torrents).
+    """
+
+    def __init__(self, session, torrent_url):
+        MyChannelBaseEndpoint.__init__(self, session)
+        self.torrent_url = torrent_url
+
+    def render_PUT(self, request):
+        """
+        Add a torrent by magnet or url to your channel. Returns error 500 if something is wrong with the torrent file
+        and DuplicateTorrentFileError if already added to your channel (except with magnet links).
+
+        Example request:
+        {
+            "description" (optional): "A video of my cat" (default: empty)
+        }
+        """
+        my_channel_id = self.channel_db_handler.getMyChannelId()
+        if my_channel_id is None:
+            return MyChannelBaseEndpoint.return_404(request)
+
+        parameters = http.parse_qs(request.content.read(), 1)
+
+        if 'description' not in parameters or len(parameters['description']) == 0:
+            extra_info = {}
+        else:
+            extra_info = {'description': parameters['description'][0]}
+
+        try:
+            if self.torrent_url.startswith("http:") or self.torrent_url.startswith("https:"):
+                torrent_def = TorrentDef.load_from_url(self.torrent_url)
+                self.session.add_torrent_def_to_channel(my_channel_id, torrent_def, extra_info, forward=True)
+            '''
```

Comment:
I'm not sure whether we want commented-out code in the PR, even if we are going to add it at one point later. The `EmercoinMgr` will probably be refactored anyway so this code may not be relevant in the future.